### DISCUSSION TITLE
JDK-11.0.4.1 +1 Installation Error on Mac

### DISCUSSION THREAD

**Discussion by MichaelBonfanti** [adoptium/adoptium-support#490] [Q&A] [answered]
Hi I have a problem installing the jdk (jdk-11.0.14.1 + 1) l-intsaller tells me that it is installed correctly but when I go to the terminal and do javac it gives me an error. the operating system I use is MacOS Catalina 10.15.7

**Comment by karianna** [comment, marked as answer]
Can you try running the full path `/Library/Java/JavaVirtualMachines/temurin-11.jdk/Contents/Home/bin/javac --version`?

Also can you tell us what:

`echo $PATH` and `echo $JAVA_HOME` are?

**Reply by karianna** [reply]
Now covered at https://github.com/adoptium/adoptium-support/issues/488

**Reply by MichaelBonfanti** [reply]
here's what he asked me for

![Schermata 2022-04-17 alle 14 57 03](https://user-images.githubusercontent.com/53140578/163715457-c1102121-84f4-497c-846e-ac716a8c6e84.png)
![Schermata 2022-04-17 alle 14 58 55](https://user-images.githubusercontent.com/53140578/163715463-4d1dffb5-6387-428d-b066-aee979883590.png)
![Schermata 2022-04-17 alle 15 00 06](https://user-images.githubusercontent.com/53140578/163715465-d6d165da-6014-4ed3-9808-de6cca69d028.png)

**Reply by karianna** [reply]
OK now try `which javac`

**Reply by MichaelBonfanti** [reply]
![Schermata 2022-04-17 alle 15 23 04](https://user-images.githubusercontent.com/53140578/163716259-11768258-fff7-45fc-aacc-3bf59e985d9a.png)

**Reply by karianna** [reply]
try `ls -la /usr/bin/javac`

**Reply by MichaelBonfanti** [reply]
![Schermata 2022-04-17 alle 17 35 49](https://user-images.githubusercontent.com/53140578/163721630-da7224ea-9cb2-418f-8547-d28ff264d22a.png)

**Comment by karianna** [comment]
Ok, ls -la that new full path and see what it really points to 🙂

On Sun, 17 Apr 2022 at 16:36, Michael Bonfanti ***@***.***>
wrote:

> [image: Schermata 2022-04-17 alle 17 35 49]
> <https://user-images.githubusercontent.com/53140578/163721630-da7224ea-9cb2-418f-8547-d28ff264d22a.png>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/adoptium/adoptium-support/discussions/490#discussioncomment-2582106>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABME2GC2RETOYYONR6IB73VFQVXTANCNFSM5TOWTB2Q>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***
> com>
>
-- 
Cheers, Martijn (Sent from Gmail Mobile)

**Reply by MichaelBonfanti** [reply]
![Schermata 2022-04-17 alle 18 19 13](https://user-images.githubusercontent.com/53140578/163723326-d35ffabd-cd7d-400a-8ada-ef61d6fb0d72.png)

**Reply by karianna** [reply]
Sorry I meant `ls -la` the full path that was returned by the `ls -la /usr/bin/javac`

**Comment by MichaelBonfanti** [comment]
![Schermata 2022-04-20 alle 15 30 45](https://user-images.githubusercontent.com/53140578/164241641-5abc76e0-daf4-44d9-bbfe-1f348c7bcdde.png)

**Reply by karianna** [reply]
OK now `ls -la /System/Library/Frameworks/JavaVM.framwework/Versions/Current/Commands/javac`

**Reply by MichaelBonfanti** [reply]
![Schermata 2022-04-21 alle 14 38 54](https://user-images.githubusercontent.com/53140578/164459729-8b0de191-c271-43cf-88dc-dfbecf94ce12.png)

**Reply by karianna** [reply]
Sorry had a typo - try:

`ls -la /System/Library/Frameworks/JavaVM.framework/Versions/Current/Commands/javac`

**Comment by MichaelBonfanti** [comment]
![Schermata 2022-04-21 alle 14 48 34](https://user-images.githubusercontent.com/53140578/164461380-1291bb4a-48b7-4410-8741-6f79ba16b707.png)

**Reply by karianna** [reply]
Hmm, I'm really not sure how this is there.  Ah that's probably Apple's old JDK.  Try running:

`export JAVA_HOME=$(/usr/libexec/java_home)`

**Reply by MichaelBonfanti** [reply]
![Schermata 2022-04-22 alle 11 34 32](https://user-images.githubusercontent.com/53140578/164679387-e8779a4b-3019-49b6-b392-928fec52b929.png)

**Comment by karianna** [comment]
Ok see if running javac works for you now 🙂

On Fri, 22 Apr 2022 at 02:35, Michael Bonfanti ***@***.***>
wrote:

> [image: Schermata 2022-04-22 alle 11 34 32]
> <https://user-images.githubusercontent.com/53140578/164679387-e8779a4b-3019-49b6-b392-928fec52b929.png>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/adoptium/adoptium-support/discussions/490#discussioncomment-2614536>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABME2DQ6AUDJ5TIKW3B6UDVGJXGBANCNFSM5TOWTB2Q>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***
> com>
>
-- 
Cheers, Martijn (Sent from Gmail Mobile)

**Comment by MichaelBonfanti** [comment]
does not work
![Schermata 2022-04-23 alle 10 37 28](https://user-images.githubusercontent.com/53140578/164887187-714b52f0-88ae-4b32-850b-2ad7b0ad7d19.png)

**Comment by karianna** [comment]
OK I see what's going on.  Your JAVA_HOME is pointing to the JRE (which doesn't contain `javac`) as opposed to the JDK (which does container `javac`.  I'd strongly recommend using https://www.jenv.be/ to manage your various versions going forwards.

**Reply by MichaelBonfanti** [reply]
> OK I see what's going on.  Your JAVA_HOME is pointing to the JRE (which doesn't contain `javac`) as opposed to the JDK (which does container `javac`.  I'd strongly recommend using https://www.jenv.be/ to manage your various versions going forwards. 

Thank you very much now it works
